### PR TITLE
Refactor several tests to reduce the time spent (issue #894)

### PR DESCRIPTION
**Pull request by hramezani**
Refactor several tests to reduce the time spent (issue #894)

### CODE REVIEW

**Comment by hramezani** [comment]
@aaugustin I don't know why `test_no_candidates` and `test_no_candidates_pre` fail in `py27-pip8.1.1` environment.

File: tests/test_cli_compile.py [hunk 1]
```diff
@@ -542,36 +541,35 @@ def test_generate_hashes_verbose(runner):
...
+    assert "small-fake-a==0.1" in out.stderr
     assert "unknown_package" not in out.stderr
 
 
-@pytest.mark.network
-def test_no_candidates(runner):
+def test_no_candidates(pip_conf, runner):
     with open("requirements", "w") as req_in:
-        req_in.write("six>1.0b0,<1.0b0")
+        req_in.write("small-fake-a==>0.3b1,<0.3b1")
```
Comment (atugushev) [inline tests/test_cli_compile.py]: Probably a typo?

```suggestion
        req_in.write("small-fake-a>0.3b1,<0.3b1")
```

File: tests/test_cli_compile.py [hunk 2]
```diff
@@ -542,36 +541,35 @@ def test_generate_hashes_verbose(runner):
...
     assert out.exit_code == 2
     assert "Skipped pre-versions:" in out.stderr
 
 
-@pytest.mark.network
-def test_no_candidates_pre(runner):
+def test_no_candidates_pre(pip_conf, runner):
     with open("requirements", "w") as req_in:
-        req_in.write("six>1.0b0,<1.0b0")
+        req_in.write("small-fake-a==>0.3b1,<0.3b1")
```
Comment (atugushev) [inline tests/test_cli_compile.py]: Probably a typo?

```suggestion
        req_in.write("small-fake-a==>0.3b1,<0.3b1")
```

**Comment by atugushev** [comment]
@hramezani don't forget to run `tox -e checkqa` before commit to avoid style/code format issues. Also, I'd recommend using https://pre-commit.com. Run once `pre-commit install` and forget about running linters manually.

File: tests/test_data/small_fake_package_a/setup.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+from setuptools import setup
+
+setup(
+    name="small_fake_with_deps_a", version=0.1, install_requires=["small-fake-a==0.1"]
```
Comment (atugushev) [inline tests/test_data/small_fake_package_a/setup.py]: The dir name should have the same name as the package name. I know the other dirs do not follow this rule, but I think we should fix it in the following PR.

Comment (atugushev) [reply tests/test_data/small_fake_package_a/setup.py]: Apologize for the confusion, I meant to change `tests/test_data/small_fake_with_deps/setup.py` to `tests/test_data/small_fake_with_deps_a/setup.py`

Comment (hramezani) [reply tests/test_data/small_fake_package_a/setup.py]: No problem. I will continue to change the other package name in other PR after this one is merged.

Comment (atugushev) [reply tests/test_data/small_fake_package_a/setup.py]: Nice! 👍

**Comment by hramezani** [comment]
@aaugustin It's Done.

**Comment by aaugustin** [comment]
Hello, it's unclear to me why you're tagging me here... Is this something we talked about before? I'm not connecting the dots :-|

**Comment by hramezani** [comment]
> Hello, it's unclear to me why you're tagging me here... Is this something we talked about before? I'm not connecting the dots :-|

Oops, Sorry about that. I wanted to mention @atugushev

**Comment by aaugustin** [comment]
No problem. There's a lot of people in the jazzband org :-)

**Comment by hramezani** [comment]
@jayvdb You are welcome!

**Pull request merged**